tests/helics/application_api/LoggingTests.cpp

```diff
@@ -257,3 +257,32 @@ TEST(logging_tests, check_log_message_levels_high)
     }
     EXPECT_TRUE(found_low && found_high);
 }
+
+TEST(logging_tests, dumplog)
+{
+    helics::FederateInfo fi(CORE_TYPE_TO_TEST);
+    fi.coreInitString = "--autobroker";
+    fi.setProperty(helics::defs::log_level, -1);
+
+    auto Fed = std::make_shared<helics::Federate>("test1", fi);
+    auto cr = Fed->getCorePointer();
+    gmlc::libguarded::guarded<std::vector<std::pair<int, std::string>>> mlog;
+    cr->setLoggingCallback(helics::local_core_id,
+                           [&mlog](int level,
+                                   const std::string& /*unused*/,
+                                   const std::string& message) {
+                               mlog.lock()->emplace_back(level, message);
+                           });
+
+    Fed->enterExecutingMode();
+
+    Fed->setFlagOption(helics_flag_dumplog);
+    Fed->setFlagOption(helics_flag_dumplog, false);
```

Comment:
What is setting the flag twice testing?

Reply:
It is turning it on and off during the run, so I only get a limited number of messages in the log file.  There is one that depending on the timing of the different threads could come in between the two but in most cases the log callback will only have two messages.  The initial log message and the dumplog message.  If it has fewer the dump log didn't work at all, if it has more than 3 the turn off didn't work.

Reply:
Okay. Should probably have a comment mentioning that reason for turning it off right after turning it on.